scheduler/reconnecting_picker/reconnecting_picker.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package reconnectingpicker
+
+import (
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type ReconnectingPicker struct {
+	logger log.Logger
+}
+
+func New(logger log.Logger) *ReconnectingPicker {
+	rp := ReconnectingPicker{
+		logger: log.L().Named("reconnecting-picker"),
```

Comment:
```suggestion
		logger: logger.Named("reconnecting-picker"),
```

I think that, if we use the global default logger, we won't have the log configurations applied. Since the reconciler is already providing a logger here we should it instead.